.pre-commit-config.yaml

```diff
@@ -75,3 +75,18 @@ repos:
     args:
     - --extend-ignore=E203,F401,F811,E501
     exclude: (ctypes_mapping\/api\.py)$
+
+# Rust tools
+- repo: https://github.com/doublify/pre-commit-rust
+  rev: v1.0
+  hooks:
+  - id: fmt
+    name: Rust-fmt-xnvme
+    args: ['--verbose', '--manifest-path', 'rust/xnvme/Cargo.toml', '--']
```

Comment:
Will this format the code in place, or will it error out if formatting is wrong? I guess the latter is what we want?

Reply:
Did a quick check. It does both. That is, when running pre-commit it will fix the format, and it will output that it needed to. E.g. like this:

```
make format-all
## xNVMe: format-all
C-format-headers.........................................................Passed
C-format-source..........................................................Passed
reuse....................................................................Passed
YAML-format-check........................................................Passed
YAML-format..............................................................Passed
Python-format-black......................................................Passed
Python-format-isort......................................................Passed
Python-lint-flake8.......................................................Passed
Rust-fmt-xnvme...........................................................Failed
- hook id: fmt
- files were modified by this hook

[bin (2021)] "/Users/safl/git/xnvme/rust/xnvme/src/main.rs"
rustfmt --edition 2021 rust/xnvme-sys/src/bin/xnvme_rust_example01.rs rust/xnvme-sys/build.rs rust/xnvme-sys/src/lib.rs rust/xnvme/src/main.rs /Users/safl/git/xnvme/rust/xnvme/src/main.rs

Rust-fmt-xnvme-sys.......................................................Passed
make: *** [format-all] Error 1
```

 Thus, it is useful for cleaning up the code and for the ci-task or checking whether cod eon a PR has formatting issues.